docs/AQL/master03-syntax.adoc

```diff
@@ -475,6 +491,72 @@ The `AVG()` function returns the average value of given _expression_ argument. T
+* *translate*: Translate a code; this will translate (find an equivalent) code from one Value Set to another one based on a predefined mapping available in the external terminology service. Translation precision may not be limited to full equivalence and different kinds of mappings may be possible (wider meaning, equivalent, narrower meaning, etc).
+* *subsumes*: Subsumption testing; this will determine if a particular terminology concept is a subtype (is-a) of another one. For example, test in SNOMED-CT if `Myasthenia Gravis | 91637004` is a subtype of `autoimmune disease | 85828009` (i.e. test if 85828009 subsumes 91637004).
+
+The `driver` argument represents an identifier of the terminology server API. It provides the information to send requests related to the ValueSet and operation parameters to the the right terminology service. In addition, it allows the system to understand how to parse responses back from the terminology service.
+Some examples of mostly used drivers are:
+
+[width="100%",cols="2,4",options="header",]
+|===
+|Driver     |Description
+
+|http://hl7.org/fhir/4.0       |FHIR Terminology Service v4.0
+|http://hl7.org/fhir/3.0       |FHIR Terminology Service v3.0
+|http://hl7.org/fhir/1.0       |FHIR Terminology Service v1.0
+|ots.oceanhealthsystems.com    |Ocean Ontology Service
+|bts.better.care               |Better Terminology Service
+|dts4.apelon.com               |
+|dts5.apelon.com               |
+|===
+
+The `params_uri` argument consists of URI path and query sections in compliance with {rfc3986}[IETF RFC 3986]. Most common operations uses this to identify the value set on which to perform the operation. A value set may be a full code system such as the full set of codes in SNOMED-CT or LOINC. The value set URI may include the version and edition sections (e.g. Australian, US, etc.). When the release and version identifiers are not provided, it is up to the external terminology server to decide which default version will be used.
+Some examples of mostly used URI parameters are:
+
+[width="100%",cols="2,4",options="header",]
+|===
+|URI parameters     |Description
+
+|http://terminology.hl7.org/ValueSet/v3-FamilyMember       |
+|https://vsac.nlm.nih.gov/valueset/2.16.840.1.113762.1.4.1010.2       |
+|http://snomed.info/sct/32506021000036107/version/20200331?fhir_vs=refset/1200161000168100       |implicit VS defined for Australian Refset for Vaccination Reason
+|http://snomed.info/sct     |
+|http://www.omim.org        |
+|http://fhir.de/CodeSystem/dimdi/icd-10-gm      |
+|===
+
+The `TERMINOLOGY` function can be used in two ways in the `WHERE` clause, as a right operand of `matches` operator:
+
+* as a direct use of function results, to indicate that the set will be resolved by the internal representation, e.g.
++
+--------
+WHERE e/value/defining_code/code_string matches TERMINOLOGY('expand', 'http://hl7.org/fhir/4.0', 'http://snomed.info/sct?fhir_vs=isa/50697003')
```

Comment:
Personally I think embedding literal terminology server ids (the driver) in queries is likely to make them fragile. What if you want to change your terminology service and you have 1000 queries? Including operational ones loaded in the Definition service? I think we should use a logical service name, e.g. 'fhir_ts' might be a terminology service that obeys the FHIR terminology service protocol.

Reply:
I think the ids is something openEHR needs to publish and maintain. In the case of the FHIR one is the id recommended by Grahame Grieve.
They are not physical addresses but "terminology server API flavors". At the end a common set of identifiers will be needed for them.